src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/DefaultGetDraftViewOperation.java

```diff
@@ -0,0 +1,102 @@
+
+    public static final String QUALIFIER = "defaultDraft";
+    static final String DRAFT_ID = "DRAFT%s";
+    private static final CaseViewTrigger DELETE_TRIGGER = aCaseViewTrigger()
+        .withName("Delete")
+        .withDescription("Delete draft")
+        .withOrder(2)
+        .build();
+    private static final String RESUME = "Resume";
+
+    private final DraftGateway draftGateway;
+
+    @Autowired
+    public DefaultGetDraftViewOperation(@Qualifier(CreatorGetCaseOperation.QUALIFIER) final GetCaseOperation getCaseOperation,
+                                        final UIDefinitionRepository uiDefinitionRepository,
+                                        final CaseTypeService caseTypeService,
+                                        final UIDService uidService,
+                                        @Qualifier(DefaultDraftGateway.QUALIFIER) final DraftGateway draftGateway) {
+        super(getCaseOperation, uiDefinitionRepository, caseTypeService, uidService);
+        this.draftGateway = draftGateway;
+    }
+
+    @Override
+    public CaseView execute(String jurisdictionId, String caseTypeId, String draftId) {
+
+        final CaseType caseType = getCaseType(jurisdictionId, caseTypeId);
+        final DraftResponse draftResponse = draftGateway.get(draftId);
+
+        final CaseDetails caseDetails = buildCaseDetailsFromDraft(draftResponse);
+        final CaseViewTrigger resumeTrigger = buildResumeTriggerFromDraft(draftResponse);
+
+        final CaseTabCollection caseTabCollection = getCaseTabCollection(getCaseTypeIdFromDraft(draftResponse));
+
+        return merge(caseDetails, resumeTrigger, caseType, caseTabCollection);
+    }
+
+    private CaseViewTrigger buildResumeTriggerFromDraft(DraftResponse draftResponse) {
+        return aCaseViewTrigger()
+            .withId(draftResponse.getDocument().getEventTriggerId())
+            .withName(RESUME)
+            .withDescription(draftResponse.getDocument().getCaseDataContent().getEvent().getDescription())
+            .withOrder(1)
+            .build();
+    }
+
+    private String getCaseTypeIdFromDraft(DraftResponse draftResponse) {
+        return draftResponse.getDocument().getCaseTypeId();
+    }
+
+    private CaseView merge(CaseDetails caseDetails, CaseViewTrigger resumeTrigger, CaseType caseType, CaseTabCollection caseTabCollection) {
+        CaseView caseView = new CaseView();
+        caseView.setCaseId(caseDetails.getId().toString());
+        caseView.setChannels(caseTabCollection.getChannels().toArray(new String[0]));
+
+        caseView.setCaseType(CaseViewType.createFrom(caseType));
+        caseView.setTabs(getTabs(caseDetails, caseDetails.getData(), caseTabCollection));
+
+        caseView.setTriggers(new CaseViewTrigger[] {resumeTrigger, DELETE_TRIGGER});
+        caseView.setEvents(new CaseViewEvent[]{});
+
+        return caseView;
+    }
+
+    private CaseDetails buildCaseDetailsFromDraft(DraftResponse draftResponse) {
```

Comment:
I think this deserves it's own mapper class

Reply:
That is a bit of a stretch for few lines.

Reply:
in fact we have a duplicated block of code in `DefaultGetDraftsOperation` that is doing the same mapping. We could introduce a mapper class and use it here and `DefaultGetDraftsOperation`